JPS_VIRTUALSENSOR/ShipInputAgent/src/main/java/com/cmclinnovations/ship/DataUploader.java

```diff
@@ -57,11 +65,39 @@ public static void uploadShips(List<Ship> ships, QueryClient queryClient) throws
         // sets course, speed, location measure IRI in ship objects to be used later
         queryClient.setMeasureIri(ships);
 
-        // add a row in RDB time series data, also updates derivation timestamps
-        queryClient.updateTimeSeriesData(ships);
+        switch (source) {
+            case JSON:
+
+                LOGGER.info("Upload ship data from JSON.");
+                // add a row in RDB time series data, also updates derivation timestamps
+                queryClient.updateTimeSeriesData(ships);
+
+                break;
+
+            case RDB:
+
+                LOGGER.info("Upload ship data from relational database.");
+                if (EnvConfig.SKIP_UPDATE_RDB) {
+                    // only update time series of new ships
+                    // skip updating existing ships
+                    queryClient.bulkUpdateTimeSeriesData(newlyCreatedShips);
+                } else {
+                    // update all ships
+                    queryClient.bulkUpdateTimeSeriesData(ships);
+                }
```

Comment:
this might not be necessary, the TimeSeriesClient ensures each time row is unique